packages/core/lib/segments/segment.d.ts

```diff
@@ -0,0 +1,59 @@
+import Subsegment = require('./attributes/subsegment');
+import IncomingRequestData = require('../middleware/incoming_request_data');
+
+declare class Segment {
+  id: string;
+  name: string;
+  start_time: number;
+  in_progress?: boolean;
+  counter: number;
+  trace_id: string;
+  parent_id?: string;
+  origin?: string;
+
+  constructor(name: string, rootId?: string | null, parentId?: string | null);
+
+  addIncomingRequestData(data: IncomingRequestData): void;
```

Comment:
the implementation sets `this.http = data`, should this class have a public `http: IncomingRequestData`?

Reply:
It would be correct to have the optional `http` field, but I am unaware of any use cases that require the http element to be accessed publicly, so I'd recommend making it private.

Reply:
Sounds good - what do you think about including private fields/methods/etc. marked as `private` vs. omitting them altogether as implementation details?

Reply:
I was thinking about that - since there are no plans to make this SDK native TS anytime soon including private fields seems like somewhat wasted work. There's certainly no harm in adding them as private fields but I will approve the PR without them.